client/src/components/StreamField/blocks/StructBlock.ts

```diff
@@ -374,7 +374,17 @@ export class StructBlock {
       // eslint-disable-next-line no-restricted-syntax
       for (const blockName in error.blockErrors) {
         if (hasOwn(error.blockErrors, blockName)) {
-          this.childBlocks[blockName].setError(error.blockErrors[blockName]);
+          const block = this.childBlocks[blockName];
+          block.setError(error.blockErrors[blockName]);
+
+          // Structural blocks have a `container` property (JQuery object),
+          // while field blocks have an `element` property (DOM element).
+          const element = block.container?.[0] || block.element;
+
+          // Trigger a 'beforematch' event on the errored block to ensure it's
+          // expanded if it's within any level of collapsible panels, including
+          // settings panels.
+          element.dispatchEvent(new Event('beforematch', { bubbles: true }));
```

Comment:
I'm not sure how we feel about dispatching the `beforematch` event on our own. This works very well, even on browsers that don't support it yet (e.g. current Safari). This is because our existing code for collapsible panels always adds the event listener for the `beforematch` event even if it's not yet supported.

Alternatively, we add another listener for something like `w-panel:open` (or `w-panel:toggle` with `detail` that forces it to open) at the same places we listen to `beforematch`, and then we dispatch that event here.

It sounds like something we'd eventually do for #10168, so I was thinking we'll just do it as part of that issue so we can figure out how the event should look.